### ISSUE TITLE
CDN Links

### ISSUE THREAD

**Issue by michaelavila** [tgriesser/knex#3355]
I maintain this little tool for tinkering with knex (http://michaelavila.com/knex-querylab/), but the knex version goes out of date. I'd like to reference the "latest" knex version (via something like https://unpkg.com/) so that the tool automatically stays up to date. The problem is that knexjs does not include build versions in the repository.

So the ask is just to include build versions of knexjs in the repository, which is what I think react does (e.g. https://reactjs.org/docs/cdn-links.html).

Is this something the team is willing to do?

Thanks.

**Comment by elhigu** [comment]
Knex does not have currently any version that is suitable to be used in browser. Having it could be useful though. I don't mind if someone starts to provide prebuilt query builder parts for browsers. Though it could be implemented easily as separate repo also.

**Comment by elhigu** [comment]
Though... we already must build that browser compatible package for documentation page, so I don't see any reason why we couldn't share that package somewhere publicly.

**Comment by michaelavila** [comment]
@elhigu yes, I think I see it bundled in here https://knexjs.org/build/bundle.js (referenced in the knexjs.org source). But, perhaps my use case (querylab) doesn't warrant making this change? Maybe I'm the only one using knexjs in the browser. I can get by with updating the website every time a new version comes out, it's just not ideal.

Let me know what you think.

**Comment by elhigu** [comment]
If someone writes support for building and publishing it in knex main repo I would take it in.

**Comment by mercmobily** [comment]
Hi there! I've been doing a comprehensive review of browser-related issues in the Knex repository, and wanted to provide an update on this request.

Back when this issue was opened in 2019, @elhigu mentioned that Knex already builds a browser-compatible package for the documentation page, and expressed willingness to accept a PR that would build and publish it as part of the main repo. The documentation site at the time used webpack to create a bundle that exposed a global `knex` variable for interactive use in the browser console.

Since then, quite a bit has changed in the browser ecosystem and in Knex's infrastructure. The documentation was moved to Vitepress, and the browser build infrastructure was removed from the main repository (webpack was removed in commit 80def448 back in 2021). The global `knex` object that used to be available on the documentation pages is no longer there, even though the documentation still references it.

Currently, Knex can be built for the browser using modern bundlers like Webpack 5, but it requires significant configuration including polyfills for Node.js core modules. I've documented the complete Webpack 5 configuration in #4474, and even created a working StackBlitz example demonstrating Knex running in the browser.

The key question for the maintainers is whether Knex should officially support CDN distribution by:

1. **Publishing pre-built browser bundles** as part of the release process (UMD or ESM format)
2. **Configuring package.json** with appropriate jsdelivr/unpkg entry points
3. **Maintaining browser build infrastructure** in the main repository
4. **Documenting browser usage patterns** and limitations

The main use cases for Knex in the browser are:
- Query building for backend APIs (generate SQL on the client, send to server)
- Educational purposes and documentation examples
- Tools like QueryLab that provide SQL query interfaces

However, there are also valid concerns about maintenance burden, bundle size (approximately 989KB), and potentially encouraging insecure patterns (like exposing database credentials in client code).

I'm consolidating all CDN-specific requests under this issue. Issue #4386 (about jsDelivr default file) is an exact duplicate and I'll be closing it in favour of this one.

**Related issues:**
- #4386 - No default jsDelivr CDN file set (exact duplicate, closing)
- #5592 - Is there any interest in running Knex in the browser?
- #6183 - docs say knex object exists on the page but it does not?
- #4474 - Not compatible with Webpack 5
- #6242 - Document bundler solutions and add clean dialect entrypoints
